packages/hash/frontend/src/lib/user.ts

```diff
@@ -47,40 +61,42 @@ export type User = MinimalUser & {
 };
 
 export const constructUser = (params: {
-  userEntityId: string;
   subgraph: Subgraph;
+  userEntityEditionId: EntityEditionId;
 }): User => {
-  const { userEntityId, subgraph } = params;
+  const { userEntityEditionId, subgraph } = params;
 
-  const outgoingHasMembershipLinks = getOutgoingLinksOfEntity({
-    entityId: userEntityId,
+  const orgMemberships = getOutgoingLinksForEntityAtMoment(
     subgraph,
-    linkTypeId: types.linkType.hasMembership.linkTypeId,
-  });
-
-  const orgMemberships = outgoingHasMembershipLinks.map(
-    ({ inner }) =>
-      subgraph.vertices[inner.targetEntityId] as unknown as EntityVertex,
+    userEntityEditionId.baseId,
+    new Date(),
+  ).filter(
+    (linkEntity) =>
+      linkEntity.metadata.entityTypeId ===
+      types.linkEntityType.orgMembership.linkEntityTypeId,
   );
 
   return {
-    ...constructMinimalUser({ userEntityId, subgraph }),
-    memberOf: orgMemberships.map(({ inner: orgMembershipEntity }) => {
-      const responsibility: string =
-        orgMembershipEntity.properties[
-          extractBaseUri(types.propertyType.responsibility.propertyTypeId)
-        ];
-
-      const outgoingOfOrgLinks = getOutgoingLinksOfEntity({
-        entityId: orgMembershipEntity.entityId,
+    ...constructMinimalUser({ userEntityEditionId, subgraph }),
+    memberOf: orgMemberships.map(({ metadata, properties }) => {
+      const responsibility: string = properties[
+        extractBaseUri(types.propertyType.responsibility.propertyTypeId)
+      ] as string;
+
+      if (!metadata.linkMetadata?.rightEntityId) {
+        throw new Error("Expected org membership to contain a right entity");
+      }
+      const org = getRightEntityForLinkEntityAtMoment(
         subgraph,
-        linkTypeId: types.linkType.ofOrg.linkTypeId,
-      });
-
-      const orgEntityId = outgoingOfOrgLinks[0]!.inner.targetEntityId;
+        metadata.editionId.baseId,
+        new Date(),
```

Comment:
This is a question, not actually a suggestion. Do we want to do this instead? Not 100% sure which time we want here

```suggestion
            /** @todo - We probably want to use entity endTime - https://app.asana.com/0/1201095311341924/1203331904553375/f */
        new Date(),
```

Reply:
What do we mean with using the `endTime` here? Would it be querying at the moment of the `endTime` or being able to filter away orgs where the membership has ended?

Reply:
Seems like we functionally always want the latest in this function so `new Date()` makes sense.